### ISSUE TITLE
prune a Unet

### ISSUE THREAD

**Issue by shockjiang** [yihui-he/channel-pruning#78]
Dear Author(s),
    I am trying to applying this channel pruning method on a Unet(which is RCNN-like network with residual branch. I have tried weeks to read your python code (Also I am quite familar with caffe's code).
    So I tried to dig out that how you pruned channels of ResNet50. However, in the release of ResNet-50-2X (https://github.com/yihui-he/channel-pruning/releases/tag/ResNet-50-2X), the code seems to use VGG instead of ResNet.  Would you please help to guide how to run your method on ResNet?

**Comment by XhqGlorry11** [comment]
@shockjiang hi,  am also considering pruning Unet using this method. Have you succeed?

**Comment by shockjiang** [comment]
not really. finally, I implement it myself, but in the method proposed in
the paper.

**Comment by XhqGlorry11** [comment]
@shockjiang GREAT!Does your implementation works?How is the result?

**Comment by shockjiang** [comment]
it's not bad. the loss decreased from 0.64 to 0.62 while the flop reduce to
78%.
